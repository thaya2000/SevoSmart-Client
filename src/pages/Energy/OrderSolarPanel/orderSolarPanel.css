.account-details-form {
    max-width: 400px;
    padding: 20px;
    border-radius: 5px;
    align-items: center; /* Center the content horizontally */
    justify-content: center; /* Center the content vertically */
   
  
  }

  .homeN{
  text-align: left;
  font-weight:500;
  font-family: 'Inter', sans-serif;
  font-size: 22px;
  margin-bottom: 20px;
  }
  .order-summery {
    text-align: center;
  }
  
  .upper-container{
    max-width: 400px;
    margin: auto;
    padding: 20px;
    border-radius: 5px;
    margin-right:0;
    align-items: center; /* Center the content horizontally */
    margin-top: 0;
    /* border:solid 2px black; */
    
    background-color: rgba(255, 255, 255);
  }
  h2 {
    font-family: 'Inter', sans-serif; 
    font-weight: 600; /* Adjust the font weight (e.g., 600 for semi-bold) */
    font-size: 32px; /* Set the font size to 64 pixels */
    margin-bottom: 35px;
  }
  
  label {
    font-family: 'Inter', sans-serif; /* Use the 'Inter' font, fallback to a sans-serif font */
    display: block;
  
    margin-bottom: 15px;
  }
  
  .form-group {
    margin-bottom: 15px;
  }
  
  label {
    display: block;
    margin-bottom: 15px;
    margin-top: 15px;
  }
  
  input {
    
    width: 100%;
    padding: 8px;
    box-sizing: border-box;
    border: 1px solid #ccc;
    border-radius: 3px;
    outline: none;
    margin-bottom: 15px;
    background-color:rgb(204, 203, 203); /* Set background color to grey */
  }
  
  
  .agreement-statementO {
    margin-top: 15px; 
    text-align: left; 
    margin-bottom: 15px;
    font-weight: 500;
    color: rgb(0, 0, 0);
  }
  
  .agreement-statement p {
    margin: 0; 
    line-height: 1.5; 
    color: #555; 
  }
  
 

.image-container {
    display:flex;
  }
  
  .image {
    width: 70%; 
    height: 25%;
    border-radius:30px;
    margin-top: 50px;
    margin-left: 70px; 
  }
  .power-up{
    text-align:center;
    color: rgb(0, 0, 0);
    font-weight: 600;
    margin-bottom: 30px;
    font-size:32px;
    font-family:'Inter',sans-serif;}

  .main-container {
    width: 50%;
  margin-right: 30px;
  border:solid 2px black;
  
 }

  .cproduct{

    font-weight: 400;
    font-family: 'Inter',sans-serif;
    font-size: 32px;
    margin-bottom: 20px; 
    
      
  }

  .help-text{
    margin-top: 20px;
    margin-bottom: 15px;
  }

  .product-image{
    border-radius: 25px;
    margin-top: 15px;
  }


  .product-name {
    text-align: left; 
    font-weight:600; 
    margin-top: 10px; 
  }
  
  .nextB {
   
    background-color: rgba(51, 75, 161, 1); 
    color: white; 
    padding: 10px 20px; 
  
    /* Hover effect styles */
    transition: background-color 0.3s ease; /* Add a smooth transition effect */
  }
  
  .nextB:hover {
    background-color: rgb(69, 73, 87); /* Change the background color on hover */
  }
  

  .help-text a {
    color: blue; /* Set the text color to blue */
    text-decoration: underline; /* Add an underline */
  }
  .energy-order-buttonB {
    display: flex;
    justify-content: center;
    width: 50%;
    height: 40px;
    padding: 10px;
    background-color: rgb(10, 64, 145);
    color: #fff;
    border: none;
    border-radius: 25px;
    cursor: pointer;
    align-self: center;
    margin-left: 70px;
    margin-top: 25px; 
    font-size: 14px;
    transition: background-color 0.3s; /* Add a smooth transition effect for the background color */
  }
  
  .energy-order-buttonB:hover {
    background-color: rgba(111, 112, 113, 0.8); /* Change the background color on hover */
  }
  
  .product-image {
    width: 300px;
    height: auto;
    transition: transform 0.3s ease; 
  }
  
  .product-image:hover {
    transform: scale(0.9); }
    @media screen and (max-width: 768px) {
      .account-details-form,
      .upper-container,
      .main-container {
        max-width: 100%;
        margin: 0;
        padding: 0;
      }
    
      .main-container {
        width: 100%;
        border: 2px black;
      }
    
      .upper-container {
        display: flex;
        flex-direction: column;
        margin: 0;
        padding: 0;
        margin-bottom: 20px;
      }
    
      .image-container {
        width: 100%;
        margin-top: 20px;
        display: flex;
        flex-direction: column;
        margin: 0; /* Reset margin */
        padding: 0; /* Reset padding */
      }
      .power-up{
        margin-left: -30px;
      }
    }
    
    @media screen and (max-width: 1200px) {
      .account-details-form,
      .upper-container,
      .main-container {
        max-width: 100%;
      }
    
      .main-container {
        margin-right: 0;
        width: 100%;
      }
    
      .upper-container {
        display: flex;
        flex-direction: column;
        align-items: center;
        
      }
    
      .image-container {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin: 0; /* Reset margin */
        padding: 0; /* Reset padding */
      }
    }
    