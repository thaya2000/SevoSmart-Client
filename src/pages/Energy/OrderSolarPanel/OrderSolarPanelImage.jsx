import { useState, useEffect } from "react";
import image1 from "../../../Images/S1.jpeg";
import image2 from "../../../Images/S2.jpeg";
import image3 from "../../../Images/S3.jpeg";
import "./orderSolarPanel.css";

function OrderSolarPanelImage() {
  const [currentImageIndex, setCurrentImageIndex] = useState(0);

  useEffect(() => {
    const images = [image1, image2, image3];

    const intervalId = setInterval(() => {
      setCurrentImageIndex((prevIndex) => (prevIndex + 1) % images.length);
    }, 3000);

    return () => clearInterval(intervalId); // Cleanup on component unmount
  }, []);

  return (
    <div className="image-container">
      <img
        src={currentImageIndex === 0 ? image1 : currentImageIndex === 1 ? image2 : image3}
        alt=""
        className="image"
      />
    </div>
  );
}

export default OrderSolarPanelImage;
