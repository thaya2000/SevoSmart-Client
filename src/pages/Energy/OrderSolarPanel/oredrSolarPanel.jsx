import React, { useState, useEffect } from 'react';
// import Link  from 'react-router-dom';
import image1 from '../../../Images/S1.jpeg';
import image2 from '../../../Images/S2.jpeg';
import image3 from '../../../Images/S3.jpeg';
import image4 from '../../../Images/S4.jpg';
import image5 from '../../../Images/S5.jpg';
import image6 from '../../../Images/S6.jpg';
import './orderSolarPanel.css';



function OrderSolarPanel() {
  

  

  const [currentImageIndex, setCurrentImageIndex] = useState(0);
 
  useEffect(() => {
    const images = [image1, image2, image3];

    const intervalId = setInterval(() => {
      setCurrentImageIndex((prevIndex) => (prevIndex + 1) % images.length);
    }, 3000);

    return () => clearInterval(intervalId); // Cleanup on component unmount
  }, []);

 
  

  return (
    <div>

    <div className="image-container">
        
      <img src={currentImageIndex === 0 ? image1 : currentImageIndex === 1 ? image2 : image3} alt="" className="image" />

      <div className="main-containerA">
        <div className="upper-container">
          {/* Content for the container above "Enter Account Details" */}
          <div className="power-up">Power Up Everything by Solar</div>
          <div className="homeN">Enter home Details</div>

          <div>
            <label htmlFor="home address">Home Address</label>
            <input
              type="text"
              id="home address"
              name="address"
              // onChange={handleChange}
              required
            />
          </div>

          <div>
            <label htmlFor="ebill">Average electric bill required per month</label>
            <input
              type="text"
              id="ebill"
              name="ebill"
              // onChange={handleChange}
              defaultValue="LKR/month"
              required
            />
          </div>
          
        <div className="account-details-formA">
          <h2 className="cproductA">Choose a product</h2>
          
            <span className="help-text">
              <a href="/help-choose-product" target="_blank" rel="noopener noreferrer"  >
                Help me choose the product
              </a>
            </span>

            <div>
              <img
                src={image4}
                alt="10 Kw Solar System Off Grid"
                className="product-image"
                style={{ width: '300px', height: 'auto' }}
              />
              <p className="product-name">10 Kw Solar System Off Grid</p>
            </div>

            <div>
              <img
                src={image5}
                alt="5Kw Solar System Off Grid"
                className="product-image"
                style={{ width: '300px', height: 'auto' }}
              />
              <p className="product-name">5 Kw Solar System Off Grid</p>
            </div>

            <div>
              <img
                src={image6}
                alt="3 Kw Solar System Off Grid"
                className="product-image"
                style={{ width: '300px', height: 'auto' }}
              />
              <p className="product-name">3 Kw Solar System Off Grid</p>
            </div>
            <div className="agreement-statementO">Panels for your existing roof with backup protection</div>

            <div className='energy-order-buttonB'>Next</div>
          
        </div>
        </div>
      </div>
    </div>
    </div>
  );
}

export default OrderSolarPanel;
