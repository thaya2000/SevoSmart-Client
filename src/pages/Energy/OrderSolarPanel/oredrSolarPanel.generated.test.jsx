import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import OrderSolarPanel from './oredrSolarPanel';

vi.mock('../../../Images/S1.jpeg');
vi.mock('../../../Images/S2.jpeg');
vi.mock('../../../Images/S3.jpeg');
vi.mock('../../../Images/S4.jpg');
vi.mock('../../../Images/S5.jpg');
vi.mock('../../../Images/S6.jpg');
vi.mock('./orderSolarPanel.css');

describe('<OrderSolarPanel>', () => {
  it('should render component', () => {
    const { asFragment } = render(<OrderSolarPanel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
