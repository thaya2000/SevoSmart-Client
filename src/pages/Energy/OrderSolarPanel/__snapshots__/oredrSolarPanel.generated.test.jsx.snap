// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<OrderSolarPanel> > should render component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="image-container"
    >
      <img
        alt=""
        class="image"
        src="/src/Images/S1.jpeg"
      />
      <div
        class="main-containerA"
      >
        <div
          class="upper-container"
        >
          <div
            class="power-up"
          >
            Power Up Everything by Solar
          </div>
          <div
            class="homeN"
          >
            Enter home Details
          </div>
          <div>
            <label
              for="home address"
            >
              Home Address
            </label>
            <input
              id="home address"
              name="address"
              required=""
              type="text"
            />
          </div>
          <div>
            <label
              for="ebill"
            >
              Average electric bill required per month
            </label>
            <input
              id="ebill"
              name="ebill"
              required=""
              type="text"
              value="LKR/month"
            />
          </div>
          <div
            class="account-details-formA"
          >
            <h2
              class="cproductA"
            >
              Choose a product
            </h2>
            <span
              class="help-text"
            >
              <a
                href="/help-choose-product"
                rel="noopener noreferrer"
                target="_blank"
              >
                Help me choose the product
              </a>
            </span>
            <div>
              <img
                alt="10 Kw Solar System Off Grid"
                class="product-image"
                src="/src/Images/S4.jpg"
                style="width: 300px; height: auto;"
              />
              <p
                class="product-name"
              >
                10 Kw Solar System Off Grid
              </p>
            </div>
            <div>
              <img
                alt="5Kw Solar System Off Grid"
                class="product-image"
                src="/src/Images/S5.jpg"
                style="width: 300px; height: auto;"
              />
              <p
                class="product-name"
              >
                5 Kw Solar System Off Grid
              </p>
            </div>
            <div>
              <img
                alt="3 Kw Solar System Off Grid"
                class="product-image"
                src="/src/Images/S6.jpg"
                style="width: 300px; height: auto;"
              />
              <p
                class="product-name"
              >
                3 Kw Solar System Off Grid
              </p>
            </div>
            <div
              class="agreement-statementO"
            >
              Panels for your existing roof with backup protection
            </div>
            <div
              class="energy-order-buttonB"
            >
              Next
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
